agent/api/appnet/client.go

```diff
@@ -13,14 +13,51 @@
 
 package appnet
 
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+)
+
+type appnetClientCtxKey int
+
 type client struct {
+	udsHttpClient http.Client
+	lock          sync.RWMutex
 }
 
-// This client is a singleton. No need to create multiple clients since the only thing that veries between requests is
-// the Appnet's admin UDS
-var defaultClient = &client{}
+const (
+	udsAddressKey   appnetClientCtxKey = iota
+	unixNetworkName                    = "unix"
+)
 
 // Client retrieves the singleton Appnet client
 func Client() *client {
-	return defaultClient
+	return &client{
+		udsHttpClient: http.Client{
+			Transport: &http.Transport{
+				DialContext: udsDialContext,
+			},
+		},
+	}
+}
+
+func udsDialContext(ctx context.Context, _, _ string) (net.Conn, error) {
+	udsPath, ok := ctx.Value(udsAddressKey).(string)
+	if !ok {
+		return nil, fmt.Errorf("appnet client: Path to appnet admin socket was not a string")
+	}
+	if udsPath == "" {
+		return nil, fmt.Errorf("appnet client: Path to appnet admin socket was blank")
+	}
+	return net.Dial(unixNetworkName, udsPath)
+}
+
+func (cl *client) GetHTTPClient() http.Client {
+	cl.lock.RLock()
+	defer cl.lock.RUnlock()
+
+	return cl.udsHttpClient
```

Comment:
I think this is the same as the comment in the other file. udsHttpClient is provided at struct creation and never changed. You can remove the lock completely since it is only used here. Also this exported function is only used internally and thus can be removed similar to the other case.

Reply:
makes sense. changed it